virtcontainers/sandbox.go

```diff
@@ -1104,6 +1104,10 @@ func (s *Sandbox) startVM() error {
 
 	s.Logger().Info("VM started")
 
+	s.agent.waitForAgentReady(s)
```

Comment:
Copying my previous comment (which I don't think is addressed in the new commits):

I'm wondering how it works with the vm factory code above. By the time we hotplug the communication channel here, it is too late for vm factory because vm factory requires the communication channel for several things: reseed the random number generator, hotplug necessary memory/cpu and online them.

Maybe we should at least check if the requested communication channel is already hotplugged in `hotplugCommunicationChannel` so that it can be called both at the vm factory and here.